### ISSUE TITLE
azure_rm.py import does not work unless NSG is attached to a VM NIC

### ISSUE THREAD

**Issue by chris-p-martin** [ansible/awx#4272]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

 - API

##### SUMMARY
Azure Dynamic Inventory import does not work if the VM does not have a Network Security Group assigned to the NIC

##### ENVIRONMENT
* AWX version: AWX 6.0.0.0
* AWX install method: docker on linux
* Ansible version:  Ansible 2.8.1 
* Operating System: Ubuntu
* Web Browser: Chrome

##### STEPS TO REPRODUCE

Create Dynamic Iventory for Azure
Import fails when no NSG is directly attached to the Network Interface Card of a VM. 

##### EXPECTED RESULTS

Inventory should still import even if there is no NSG attached to the NIC.  Attaching an NSG to a NIC is an optional configuration from an Azure point of view. 

##### ACTUAL RESULTS

Import failed. AWX errored stating security_group key was missing

##### ADDITIONAL INFORMATION

`SEARCH
    1.964 INFO     Updating inventory 2: azure
    2.357 INFO     Reading Ansible inventory source: /tmp/awx_11_l8m7kgpr/azure_rm.yml
    2.359 INFO     Using VIRTUAL_ENV: /var/lib/awx/venv/ansible
    2.359 INFO     Using PATH: /var/lib/awx/venv/ansible/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    2.359 INFO     Using PYTHONPATH: /var/lib/awx/venv/ansible/lib/python2.7/site-packages:
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/bin/awx-manage", line 11, in <module>
    load_entry_point('awx==6.0.0.0', 'console_scripts', 'awx-manage')()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/__init__.py", line 140, in manage
    execute_from_command_line(sys.argv)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1148, in handle
    raise exc
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1038, in handle
    venv_path=venv_path, verbosity=self.verbosity).load()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 214, in load
    return self.command_to_json(base_args + ['--list'])
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 197, in command_to_json
    self.method, proc.returncode, stdout, stderr))
RuntimeError: ansible-inventory failed (rc=1) with stdout:
stderr:
ansible-inventory 2.8.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/var/lib/awx/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-inventory
  python version = 2.7.5 (default, Jun 20 2019, 20:27:34) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
Using /etc/ansible/ansible.cfg as config file
 [WARNING]:  * Failed to parse /tmp/awx_11_l8m7kgpr/azure_rm.yml with auto
plugin: No key or key resulted empty for security_group in host myAzureServer,
invalid entry
  File "/usr/lib/python2.7/site-packages/ansible/inventory/manager.py", line 267, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/inventory/auto.py", line 58, in parse
    plugin.parse(inventory, loader, path, cache=cache)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/inventory/azure_rm.py", line 265, in parse
    self._get_hosts()
  File "/usr/lib/python2.7/site-packages/ansible/plugins/inventory/azure_rm.py", line 346, in _get_hosts
    self._add_host_to_keyed_groups(constructable_config_keyed_groups, h.hostvars, inventory_hostname, strict=constructable_config_strict)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/inventory/__init__.py", line 442, in _add_host_to_keyed_groups
    raise AnsibleParserError("No key or key resulted empty for %s in host %s, invalid entry" % (keyed.get('key'), host))
 [WARNING]: Unable to parse /tmp/awx_11_l8m7kgpr/azure_rm.yml as an inventory
source
ERROR! No inventory was parsed, please check your configuration and options.`

**Comment by ryanpetrello** [comment]
Looks like this might be related to https://github.com/ansible/ansible/issues/56030?

**Comment by chris-p-martin** [comment]
It seems to be a slightly different issue. I've tried to amend the script directly within my running AWX container image to see if I can fix this issue but I'm not sure AWX is picking up the changes I've made. It still reports the same error. It also doesn't honour the configuration I've made in issue 4273.

Is there something else I need to do on top of editing the script in /usr/lib/python2.7/site-packages/ansible/plugins/inventory?
